sample-app/src/main/java/com/joinforage/android/example/ui/pos/POSViewModel.kt

```diff
@@ -81,11 +86,15 @@ class POSViewModel : ViewModel() {
         }
     }
 
-    fun checkEBTCardBalance(foragePINEditText: ForagePINEditText, paymentMethodRef: String, onSuccess: (response: BalanceCheck?) -> Unit) {
+    fun checkEBTCardBalance(foragePinEditText: ForagePINEditText, paymentMethodRef: String, onSuccess: (response: BalanceCheck?) -> Unit) {
+        if (terminalId == null) {
+            throw Error("Invalid POS Terminal ID")
+        }
+
         viewModelScope.launch {
-            val response = ForageSDK().checkBalance(
+            val response = ForageTerminalSDK(terminalId!!).checkBalance(
```

Comment:
siiiick

Reply:
if it ends up being OK that `terminalId` can be a `String` and not a `String?` then we could remove the `!!` but no pressure if there's an unfavorable tradeoff